NavigationRail/src/main/java/com/microsoft/device/display/samples/navigationrail/ui/view/ItemDetailsDrawer.kt

```diff
@@ -57,18 +63,42 @@ private const val LONG_DETAILS_LINE_HEIGHT = 32f
 @ExperimentalMaterialApi
 @Composable
 fun BoxWithConstraintsScope.ItemDetailsDrawer(
-    modifier: Modifier,
     image: Image,
     isDualLandscape: Boolean,
-    foldSize: Dp,
+    isDualPortrait: Boolean,
+    foldOccludes: Boolean,
+    foldBounds: Rect,
+    windowHeight: Dp,
     gallerySection: GallerySections?,
 ) {
+    val isPortrait = LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT
+
     // Set max/min height for drawer based on orientation
-    val expandedHeightPct = if (isDualLandscape) EXPANDED_HEIGHT_2PANE else EXPANDED_HEIGHT_1PANE
-    val collapsedHeightPct = if (isDualLandscape) COLLAPSED_HEIGHT_2PANE else COLLAPSED_HEIGHT_1PANE
-    val fullHeight = constraints.maxHeight.toFloat()
-    val expandedHeight = with(LocalDensity.current) { (expandedHeightPct * fullHeight).toDp() }
-    val collapsedHeight = with(LocalDensity.current) { (collapsedHeightPct * fullHeight).toDp() }
+    val expandedHeightPct: Float
+    val collapsedHeightPct: Float
+    when {
+        isDualLandscape -> {
+            expandedHeightPct = EXPANDED_HEIGHT_2PANE
+            collapsedHeightPct = COLLAPSED_HEIGHT_2PANE
+        }
+        isDualPortrait || isPortrait -> {
+            expandedHeightPct = EXPANDED_HEIGHT_1PANE_PORTRAIT
+            collapsedHeightPct = COLLAPSED_HEIGHT_1PANE_PORTRAIT
+        }
+        else -> {
+            expandedHeightPct = EXPANDED_HEIGHT_1PANE_LANDSCAPE
+            collapsedHeightPct = COLLAPSED_HEIGHT_1PANE_LANDSCAPE
+        }
+    }
+    val foldBoundsDp: RectF
```

Comment:
Will this part go away after we move to WindowState alpha02?

Reply:
Yes, it can just be replaced by "foldBoundsDp"!